pkg/controller/controller_utils.go

```diff
@@ -289,12 +289,12 @@ func (r RealPodControl) createPods(nodeName, namespace string, template *api.Pod
 			GenerateName: prefix,
 		},
 	}
-	if len(nodeName) != 0 {
-		pod.Spec.NodeName = nodeName
-	}
 	if err := api.Scheme.Convert(&template.Spec, &pod.Spec); err != nil {
 		return fmt.Errorf("unable to convert pod template: %v", err)
 	}
+	if len(nodeName) != 0 {
+		pod.Spec.NodeName = nodeName
+	}
```

Comment:
Yes. Pods had to be scheduled by the scheduler. Your "does daemon pod exist for node" check returned false until the scheduler was done => many new daemon pods are created meanwhile.